### PR TITLE
Refactor Facility Export & Add New Sidebar collapse & expand Test

### LINKED ISSUE #9043: Refactor the Facility Export existing cypress test and Add New Sidebar collapse test along with it
**Description**

- [ ] Refactor the existing facility export by combining two existing tests and adding test for verifying sidebar collapse functionality

**General Instruction**

- Use all reusable functions from the commands.ts file and other existing tests.
- To understand if a custom function used in other existing tests exists, search using their element ID in the codebase.
- Use the POM approach; refer to the Patient_Spec folder for guidance.
- Add necessary API verifications and verify notifications and text to avoid flaky tests.
- Add the test file to FacilityHomepage.cy.ts and the related function in the facilityHome.ts file

**Task - 1: Add new test for sidebar collapse**

- Once the homepage loads, verify that all sidebar icons with their corresponding text are visible.
- Next, collapse the sidebar and confirm that only the icons are visible.
- now again expand the sidebar and verify all the sidebar icon with their corresponding text are visible

**Task - 2: Refactor the existing cypress facility export** 

Image

- Remove the "Verify Capacity Export Functionality" test, and incorporate those test lines into the existing "Verify Facility Export Functionality" test.
- Add API verification for the capacity export test as well.

### PR DESCRIPTION
**Pull request by ghost**
## Proposed Changes

- Fixes #9043
- Task - 1 : Added new test for sidebar collapse
- Task - 2: Refactored the existing cypress facility export

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

## Summary by CodeRabbit

- **New Features**
	- Introduced a test case to verify the sidebar's collapse and expand functionality.
	- Enhanced sidebar interaction with new methods for toggling and verifying visibility of icons and text.

- **Improvements**
	- Added `data-testid` attributes to components for better testability.
	- Streamlined CSV download functionality tests to include capacity exports.

- **Bug Fixes**
	- Ensured local storage is saved after each test execution.

### CODE REVIEW

**Review by Jacobjeevan — APPROVED** [review]
LGTM

File: cypress/e2e/facility_spec/FacilityHomepage.cy.ts [hunk 1]
```diff
@@ -136,6 +138,22 @@ describe("Facility Homepage Function", () => {
     facilityHome.verifyLiveMonitorUrl();
   });
 
+  it("Verify sidebar collapse and expand functionality", () => {
+    facilityHome.toggleSidebar();
+    facilityHome.verifyIconsVisible();
+    facilityHome.verifyTextVisible();
```
Comment (nihal467) [inline cypress/e2e/facility_spec/FacilityHomepage.cy.ts]: - it does not verify that each icon has a respective text next to it.

File: cypress/e2e/facility_spec/FacilityHomepage.cy.ts [hunk 2]
```diff
@@ -136,6 +138,22 @@ describe("Facility Homepage Function", () => {
...
+    // Click toggle button to collapse sidebar, verify icons visible and text hidden
+    facilityHome.toggleSidebar();
+    facilityHome.verifyIconsVisible();
+    facilityHome.verifyTextHidden();
+
+    // Click toggle button again to expand sidebar, verify icons and text are visible again
+    facilityHome.toggleSidebar();
+    facilityHome.verifyIconsVisible();
+    facilityHome.verifyTextVisible();
+
```
Comment (nihal467) [inline cypress/e2e/facility_spec/FacilityHomepage.cy.ts:155]: - once the test is complete, the brackets are missing, syntax error

File: cypress/e2e/facility_spec/FacilityHomepage.cy.ts [hunk 3]
```diff
@@ -185,3 +203,7 @@ describe("Facility Homepage Function", () => {
     cy.saveLocalStorage();
   });
 });
+
+afterEach(() => {
```
Comment (nihal467) [inline cypress/e2e/facility_spec/FacilityHomepage.cy.ts]: - delete the duplicate aftereach function

**Review by nihal467 — CHANGES_REQUESTED** [review]

File: cypress/e2e/facility_spec/FacilityHomepage.cy.ts (hunk 1 above)
Comment (ghost) [reply cypress/e2e/facility_spec/FacilityHomepage.cy.ts]: @nihal467 sir I have made changes for this & sorry for those syntax & duplicate errors I tried to fix merge conflicts directly & made those mistakes

File: cypress/pageobject/Facility/FacilityHome.ts [hunk 4]
```diff
@@ -104,6 +107,33 @@ class FacilityHome {
...
+  ) {
+    this.sidebarItems()
+      .should("have.length.at.least", 1)
+      .each(($item) => {
+        cy.wrap($item, { timeout: 10000 })
+          .find('[data-testid="sidebar-icon"]')
+          .should("be.visible");
+        cy.wrap($item, { timeout: 10000 })
+          .find('[data-testid="sidebar-text"]')
+          .should(textVisibility);
```
Comment (nihal467) [inline cypress/pageobject/Facility/FacilityHome.ts]: As mentioned earlier, ensure that the exact text is verified. For example, confirm that the text corresponding to the facility icon is precisely "Facilities" and not just any visible text. Similarly, verify the exact text for other icons as well.

**Review by nihal467 — CHANGES_REQUESTED** [review]

**Comment by nihal467** [comment]
@Alokih make the requested changes

**Comment by ghost** [comment]
@nihal467 I have my end sem exams going on currently. Can I do it after few days ?

**Comment by nihal467** [comment]
@Alokih that's fine, but can you mention a date when you are expecting to complete it

**Comment by ghost** [comment]
> @Alokih that's fine, but can you mention a date when you are expecting to complete it

Tomorrow is the last one , so I will try to complete it by Day after tomorrow.

**Comment by ghost** [comment]
@nihal467 I have made some changes as you said , can you review it once ?

**Comment by nihal467** [comment]
@Alokih your test is failing in CI/CD fix it

**Comment by ghost** [comment]
@nihal467 I am not getting any idea how to fix this, locally all tests are passing , can you help me a bit here ?

**Comment by nihal467** [comment]
@Alokih where are you stuck, did you identified the issue ?

**Comment by ghost** [comment]
I am unable to identify what the issue is ?

**Comment by nihal467** [comment]
@Alokih first identify where it is failing, the failure screenshot are accessible here : https://github.com/ohcnetwork/care_fe/actions/runs/12112894252?pr=9066

**Comment by ghost** [comment]
@nihal467 issue is in facility export test but that only I am unable to understand why it is failing in CI/CD because locally everything is passing succesfully

**Comment by nihal467** [comment]
> @nihal467 issue is in facility export test but that only I am unable to understand why it is failing in CI/CD because locally everything is passing succesfully

I understand that it’s failing in the CI/CD pipeline. When you investigated, what did you find? If the tests passed when the capacity export was kept separate, it could indicate an issue with the function itself or another underlying cause. Possible reasons might include the capacity export taking longer to generate the report, multiple requests being fired, or the backend not responding as expected.

Please investigate the issue at the component level and start a thread in the channel to share your findings after completing a thorough investigation why the cypress is failing in CI/CD for capacity export alone

**Comment by nihal467** [comment]
@Alokih, can you please share the current status of the PR? Otherwise, in the next 48 hours, the PR will be closed, and you will be unassigned from the issue due to inactivity.

**Comment by ghost** [comment]
Sorry @nihal467 I haven't made any progress here, you can close the PR & unassign me from this issue.

**Pull request closed without merging**